### ISSUE TITLE
Test email not working after save

### ISSUE THREAD

**Issue by jaishreeem** [magepal/magento2-gmailsmtpapp#25]
Hi All,
I able sent test email before press "Save Config" button. But test email not working after press "Save Config" button and received invalid username and password message. Store email also not sent.

Details:
Magento 2.1.1
MySQL 5.7.14
PHP 7.0.10

I have already reviewed below link. Suspect, not applicable for Magento 2.1.1
https://github.com/magepal/magento2-gmailsmtpapp/issues/7

Kindly help to resolve the issue. #

**Comment by srenon** [comment]
@jaishreeem ... Can you check to see what password magento is return after save .... (edit the controller to return the actual password

**Comment by jaishreeem** [comment]
@srenon ,

I have changed below line the file. I have observed that password return after save is NULL (blank).

Kindly advise whether is this the modification you expect or guide on modification.

File: Controller/Adminhtml/Test/Index.php
Original line: $result = __($e->getMessage());
Modified line: $result = __($e->getMessage()) . " u: {$username} p: {$password} - END";

**Comment by srenon** [comment]
Is this in a store view or global view? (are you run multi-store?)

The above code should have return your password

**Comment by chris-pook** [comment]
I have the same issue of test email not sending. Possibly different reason to above but it seems the password filed is not decrypted when you retrieve it in the helper

Maybe related to this: https://github.com/magento/magento2/issues/2984

If I hard code my password in there instead of using the config value my emails start sending. Might be worth adding a note to the README that google will warn of unsecure account access from an unknown app when using the default settings of the module.

**Comment by srenon** [comment]
@chris-pook ... what version of magento are you using? 

The wrong decryption of password should have been fix in magento https://github.com/magepal/magento2-gmailsmtpapp/issues/7

**Comment by chris-pook** [comment]
This was in 2.1.2

**Comment by jaishreeem** [comment]
@srenon 
Thanks for the guidance.
I have observed that some issue in Magento cache. Issue resolved after performed below action without hard code password in file.

Action taken:
Magento Admin > System [Side Menu] > Cache Management [Sub Menu] > Flush cache storage [Button].


@chris-pook 
I have observed that google accept SMTP connection without secuirty issue after used open certificate authority (like letsecrypt) and Google 2 step verification (App specific password).
